frontend/utils/time.js

```diff
@@ -87,3 +91,17 @@ export function getTime (date: Date): string {
   const t = new Date(date)
   return `${t.getHours()}:${t.getMinutes()}`
 }
+
+// Returns months passed + the ratio of (the current day of the month) / (days in current month);
+// The current day / month are calculated with respect to the atDate parameter. Months passed
+// is caclculated from the startDate parameter.
+export function cycleAtDate (atDate: string | Date, startDate: string | Date): number {
```

Comment:
The second parameter `startDate` is causing so much confusion in the code, and bugs. Please remove it and to compare months, compare the `atDate` directly to another date using `compareMonthstamps`, or create a similar function to compare ISOStrings. I think you might even be able to write something like `new Date(isoString1) < new Date(isoString2)`